.github/workflows/ci.yml

```diff
@@ -29,27 +29,61 @@ jobs:
           - 3.7
           - 3.8
           - 3.9
+        with-s3:
+          - 1
+          - 0
```

Comment:
What if AWS SDK was not installed but we compile torchdata with `BUILD_S3=1`?

Reply:
Compiling with `BUILD_S3=1` and without AWS SDK will result in compilation failure.